lib/include/irritator/core.hpp

```diff
@@ -5220,14 +5220,76 @@ struct constant
     }
 };
 
-struct filter
-{
+struct filter {
     port x[1];
     port y[1];
-    time sigma = time_domain<time>::infinity;
-    double current_value = 0.0;
-    double up_threshold = 0.0;
-    double down_threshold = 0.0;
+    time sigma;
+
+    double default_lower_threshold;
+    double default_upper_threshold;
+
+    double lower_threshold;
+    double upper_threshold;
+
+    double default_value;
+    double default_input_coeff;
+
+    double value;
+    double input_coeff;
```

Comment:
Why input_coeff? Filter model only check if the current message value is between default_lower_threshold and default_upper_threshold.